src/gateways/GatewayKeyValuePairConfidentialValueFields.js

```diff
@@ -0,0 +1,62 @@
+import {
+    InputFieldFF,
+    ReactFinalForm,
+    composeValidators,
+    string,
+    hasValue,
+} from '@dhis2/ui'
+import { PropTypes } from '@dhis2/prop-types'
+import React from 'react'
+
+import {
+    FIELD_GATEWAY_KEY_VALUE_PAIR_VALUE_LABEL,
+    GatewayKeyValuePairValueField,
+    createFieldGatewayKeyValuePairValueName,
+} from './GatewayKeyValuePairValueField'
+import { FormRow, createEqualTo } from '../forms'
+import { dataTest } from '../dataTest'
+import i18n from '../locales'
+
+const { Field } = ReactFinalForm
+
+const createEqualToConfidentialValue = index =>
+    createEqualTo(
+        console.log(
+            'createFieldGatewayKeyValuePairValueName(index)',
+            createFieldGatewayKeyValuePairValueName(index)
+        ) || createFieldGatewayKeyValuePairValueName(index),
+        FIELD_GATEWAY_KEY_VALUE_PAIR_VALUE_LABEL
```

Comment:
Pretty sure we would want to get rid of the console.log here...

Reply:
yep...

Reply:
removed